File: Data Structures & Algorithms/generate-parentheses/submission-1.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        stack = []
        res = []

        def backtrack(Open, Close):
            if Open == Close == n:
                res.append("".join(stack))
                return
            if Open < n:
                stack.append('(')
                backtrack(Open + 1, Close)
                stack.pop()
            if Close < Open:
                stack.append(')')
                backtrack(Open, Close + 1)
                stack.pop()
            
            return
        

        backtrack(0,0)

        return res
        

File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<int, int> mp;
        stack<int> st;

        if (s.size() != t.size()) return false;

        for (int i = 0; i < s.size(); i++) {
            mp[s[i]]++;
            st.push(t[i]);
        }

        for (int i = 0; i < s.size(); i++) {
            if (mp[st.top()]) {
                mp[st.top()]--;
                st.pop();
            }
        }
        return st.empty();
    }
};
